runtime/io-api.h

```diff
@@ -0,0 +1,308 @@
+bool IONAME(SetRound)(Cookie, const char *, std::size_t);
+// SIGN=PLUS, SUPPRESS, PROCESSOR_DEFINED
+bool IONAME(SetSign)(Cookie, const char *, std::size_t);
+
+// Data item transfer for modes other than namelist.
+// Any data object that can be passed as an actual argument without the
+// use of a temporary can be transferred by means of a descriptor;
+// vector-valued subscripts and coindexing will require elementwise
+// transfers &/or data copies.  Unformatted transfers to/from contiguous
+// blocks of local image memory can avoid the descriptor, and there
+// are specializations for the most common scalar types.
+//
+// These functions return false when the I/O statement has encountered an
+// error or end-of-file/record condition that the caller has indicated
+// should not cause termination of the image by the runtime library.
+// Once the statement has encountered an error, all following items will be
+// ignored and also return false; but compiled code should check for errors
+// and avoid the following items when they might crash.
+bool IONAME(OutputDescriptor)(Cookie, const Descriptor &);
+bool IONAME(InputDescriptor)(Cookie, const Descriptor &);
+bool IONAME(OutputUnformattedBlock)(Cookie, const char *, std::size_t);
+bool IONAME(InputUnformattedBlock)(Cookie, char *, std::size_t);
+bool IONAME(OutputInteger64)(Cookie, std::int64_t);
+bool IONAME(InputInteger64)(Cookie, std::int64_t &, int kind = 8);
+bool IONAME(OutputReal32)(Cookie, float);
+bool IONAME(InputReal32)(Cookie, float &);
+bool IONAME(OutputReal64)(Cookie, double);
+bool IONAME(InputReal64)(Cookie, double &);
+bool IONAME(OutputComplex32)(Cookie, float, float);
+bool IONAME(OutputComplex64)(Cookie, double, double);
+bool IONAME(OutputAscii)(Cookie, const char *, std::size_t);
+bool IONAME(InputAscii)(Cookie, char *, std::size_t);
+bool IONAME(OutputLogical)(Cookie, bool);
+bool IONAME(InputLogical)(Cookie, bool &);
+
+// SIZE= on a data transfer, thus far, or INQUIRE(IOLENGTH=)
+bool IONAME(GetSize)(Cookie, std::int64_t, int kind = 8);
+
+// Additional specifier interfaces for the connection-list of
+// on OPEN statement (only).  SetBlank(), SetDecimal(),
+// SetDelim(), GetIoMsg(), SetPad(), SetRound(), & SetSign()
+// are also acceptable for OPEN.
+// ACCESS=SEQUENTIAL, DIRECT, STREAM
+bool IONAME(SetAccess, Cookie, const char *, std::size_t);
+// ACTION=READ, WRITE, or READWRITE
+bool IONAME(SetAction, Cookie, const char *, std::size_t);
+// ASYNCHRONOUS=YES, NO
+bool IONAME(SetAsynchronous, Cookie, const char *, std::size_t);
+// ENCODING=UTF-8, DEFAULT
+bool IONAME(SetEncoding, Cookie, const char *, std::size_t);
+// FORM=FORMATTED, UNFORMATTED
+bool IONAME(SetForm, Cookie, const char *, std::size_t);
+// POSITION=ASIS, REWIND, APPEND
+bool IONAME(SetPosition, Cookie, const char *, std::size_t);
+bool IONAME(SetRecl, Cookie, std::size_t);  // RECL=
+// STATUS=OLD, NEW, SCRATCH, REPLACE, UNKNOWN
+bool IONAME(SetStatus, Cookie, const char *, std::size_t);
```

Comment:
SetStatus can probably also be used for CLOSE.  The set of permitted values for CLOSE is not the same as for OPEN.